Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py

```diff
@@ -0,0 +1,1842 @@
+                                       headers=table_headers, removeNull=True)
+    return CommandResults(outputs_key_field='id', outputs_prefix='SophosCentral.DetectedExploit',
+                          raw_response=results, outputs=outputs, readable_output=readable_output)
+
+
+def sophos_central_detected_exploit_get_command(client: Client, args: dict) -> CommandResults:
+    """
+    Get a single detected exploit.
+
+    Args:
+        client (Client): Sophos Central API client.
+        args (dict): All command arguments
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    """
+    detected_exploit_id = args.get('detected_exploit_id', '')
+    try:
+        result = client.get_detected_exploit(detected_exploit_id)
+        table_headers = ['id', 'description', 'thumbprint', 'count', 'firstSeenAt', 'lastSeenAt']
+        object_data = {}
+        if result:
+            object_data = create_detected_exploit_output(result)
+
+        readable_output = tableToMarkdown(name='Found Detected Exploit:', t=object_data,
+                                          headers=table_headers, removeNull=True)
+        return CommandResults(outputs_key_field='id',
+                              outputs_prefix='SophosCentral.DetectedExploit',
+                              raw_response=result, outputs=object_data,
+                              readable_output=readable_output)
+    except DemistoException:
+        return CommandResults(f'Unable to find exploit: {detected_exploit_id}')
+
+
+def fetch_incidents(client: Client, last_run: Dict[str, int],
+                    first_fetch_time: str, fetch_severity: Optional[List[str]],
+                    fetch_category: Optional[List[str]],
+                    max_fetch: Optional[int]) -> Tuple[Dict[str, int], List[dict]]:
+    """
+    Fetch incidents (alerts) each minute (by default).
+    Args:
+        client (Client): Sophos Central Client.
+        last_run (dict): Dict with last_fetch object,
+                                  saving the last fetch time(in millisecond timestamp).
+        first_fetch_time (dict): Dict with first fetch time in str (ex: 3 days ago).
+        fetch_severity (list(str)): Severity to fetch.
+        fetch_category (list(str)): Category(s) to fetch.
+        max_fetch (int): Max number of alerts to fetch.
+    Returns:
+        Tuple of next_run (millisecond timestamp) and the incidents list
+    """
+    last_fetch_timestamp = last_run.get('last_fetch', None)
+
+    if last_fetch_timestamp:
+        last_fetch_date = datetime.fromtimestamp(last_fetch_timestamp / 1000)
+        last_fetch = last_fetch_date
+    else:
+        first_fetch_date = dateparser.parse(first_fetch_time).replace(tzinfo=None)
+        last_fetch = first_fetch_date
+
+    incidents = []
+    next_run = last_fetch
+    alerts = client.search_alert(last_fetch.strftime(DATE_FORMAT)[:-4] + 'Z', None, None,

```

Comment:
we prefer using `dateparser` over strftime due to performance
https://xsoar.pan.dev/docs/integrations/code-conventions#fetch-incidents

Reply:
Can dateparser convert datetime to string?

Reply:
yes. you can also use `parse_date_range` if you wish.
in any case please do not manually split/cut/trim the given string representing the time.